acme/tests/client_test.py

```diff
@@ -88,633 +86,6 @@ def setUp(self):
-        self.response.content = CERT_DER
-        self.assertEqual(self.certr.update(body=messages_test.CERT),
-                         self.client.check_cert(self.certr))
-
-        # TODO: split here and separate test
-        self.response.headers['Location'] = 'foo'
-        self.assertRaises(
-            errors.UnexpectedUpdate, self.client.check_cert, self.certr)
-
-    def test_check_cert_missing_location(self):
-        self.response.content = CERT_DER
-        self.assertRaises(
-            errors.ClientError, self.client.check_cert, self.certr)
-
-    def test_refresh(self):
-        self.client.check_cert = mock.MagicMock()
-        self.assertEqual(
-            self.client.check_cert(self.certr), self.client.refresh(self.certr))
-
-    def test_fetch_chain_no_up_link(self):
-        self.assertEqual([], self.client.fetch_chain(self.certr.update(
-            cert_chain_uri=None)))
-
-    def test_fetch_chain_single(self):
-        # pylint: disable=protected-access
-        self.client._get_cert = mock.MagicMock()
-        self.client._get_cert.return_value = (
-            mock.MagicMock(links={}), "certificate")
-        self.assertEqual([self.client._get_cert(self.certr.cert_chain_uri)[1]],
-                         self.client.fetch_chain(self.certr))
-
-    def test_fetch_chain_max(self):
-        # pylint: disable=protected-access
-        up_response = mock.MagicMock(links={'up': {'url': 'http://cert'}})
-        noup_response = mock.MagicMock(links={})
-        self.client._get_cert = mock.MagicMock()
-        self.client._get_cert.side_effect = [
-            (up_response, "cert")] * 9 + [(noup_response, "last_cert")]
-        chain = self.client.fetch_chain(self.certr, max_length=10)
-        self.assertEqual(chain, ["cert"] * 9 + ["last_cert"])
-
-    def test_fetch_chain_too_many(self):  # recursive
-        # pylint: disable=protected-access
-        response = mock.MagicMock(links={'up': {'url': 'http://cert'}})
-        self.client._get_cert = mock.MagicMock()
-        self.client._get_cert.return_value = (response, "certificate")
-        self.assertRaises(errors.Error, self.client.fetch_chain, self.certr)
-
-    def test_revoke(self):
-        self.client.revoke(self.certr.body, self.rsn)
-        self.net.post.assert_called_once_with(
-            self.directory[messages.Revocation], mock.ANY, acme_version=1)
-
-    def test_revocation_payload(self):
-        obj = messages.Revocation(certificate=self.certr.body, reason=self.rsn)
-        self.assertIn('reason', obj.to_partial_json().keys())
-        self.assertEqual(self.rsn, obj.to_partial_json()['reason'])
-
-    def test_revoke_bad_status_raises_error(self):
-        self.response.status_code = http_client.METHOD_NOT_ALLOWED
-        self.assertRaises(
-            errors.ClientError,
-            self.client.revoke,
-            self.certr,
-            self.rsn)
-
-
 class ClientV2Test(ClientTestBase):
```

Comment:
Mildly silly to keep around `ClientTestBase`, but maybe not silly enough to be worth getting rid of.

Reply:
Yep, well spotted, thanks.